docker/internal/tarfile/readable.go

```diff
@@ -0,0 +1,227 @@
+package tarfile
+
+import (
+	"archive/tar"
+	"encoding/json"
+	"io"
+	"io/ioutil"
+	"os"
+	"path"
+
+	"github.com/containers/image/v5/internal/iolimits"
+	"github.com/containers/image/v5/internal/tmpdir"
+	"github.com/containers/image/v5/pkg/compression"
+	"github.com/containers/image/v5/types"
+	"github.com/pkg/errors"
+)
+
+// ReadableArchive is a ((docker save)-formatted) tar archive that allows random access to any component.
+type ReadableArchive struct {
```

Comment:
This name feels a bit weird, but so does `internal/tarfile.Reader`. I’m very much open to suggestions.

OTOH it’s internal, so it’s not _as_ critical.

Reply:
Trying `tarfile.Reader`, with variable name `archive`. The `internal/tarfile.Reader`/`archive.Reader` pair is somewhat symmetrical, but it might be too confusing; I’m not sure.